pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshiftVersion"`
+	BaseDnsDomain            string `json:"baseDnsDomain,omitempty"`
+	ClusterNetworkCidr       string `json:"clusterNetworkCidr,omitempty"`
+	ClusterNetworkHostPrefix int64  `json:"clusterNetworkHostPrefix,omitempty"`
+	ServiceNetworkCidr       string `json:"serviceNetworkCidr,omitempty"`
+	ApiVip                   string `json:"apiVip,omitempty"`
+	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
+	IngressVip               string `json:"ingresVip,omitempty"`
+	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
+	SshPublicKey             string `json:"sshPublicKey,omitempty"`
+	VipDhcpAllocation        bool   `json:"vipDhcpAllocation,omitempty"`
+	HttpProxy                string `json:"httpProxy,omitempty"`
+	HttpsProxy               string `json:"httpsProxy,omitempty"`
+	NoProxy                  string `json:"noProxy,omitempty"`
```

Comment:
Could NoProxy be assumed by the absence of HttpProxy and HttpsProxy?

Reply:
This field is based on a commonly-used convention born out of libcurl, where you can list domain names in this field that will not be proxied even if there are proxy settings in place. It's often used to prevent local traffic from being sent through a proxy.

Reply:
Ack!